### PR TITLE
Use eslint

### LINKED ISSUE #250: try out typescript-eslint
from recent news, tslint will be deprecated in favor of typescript-eslint and community has started moving to that package. We can make a list of rules we're using, and check them off once they're migrated, (if something's not migrated, we can send them PR), our custom rules also might be ported as well :)

Not every rule has been ported, maybe we can use both at once? At least the indentation rule might work.

### PR DESCRIPTION
**Pull request by paibamboo**
Closes #250

### CODE REVIEW

**Comment by cyberhck** [comment]
I have a few concerns:
- we can't extend with the things we were extending before? (tslint:latest and tslint-react) I think aren't applied
- were ALL rules available? If not we should make a list and watch them, once they're available we should probably add them as well
- couldn't we have used both tslint and eslint and slowly move rules gradually? That'd mean all rules would still be in effect.
- what's gonna happen to the rules we wrote in tslint-rules?

I think we need a tracking issue which watches all rules while they're being converted, and add them here.

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+                    "no-unnecessary-local-variable": true,
+                    "no-unnecessary-override": true,
+                    "no-unused-expression": true,
+                    "no-with-statement": true,
+                    "object-literal-sort-keys": true,
+                    "one-line": [
+                        true,
+                        "check-open-brace",
+                        "check-catch",
+                        "check-else",
```
Comment (cyberhck) [inline .eslintrc.js]: what does check-else mean? is it that for any if there must be an else? That'd be a bad idea though, kinda want to write as much as possible without the else part. What's `one-line`?

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+                    "node_modules/@crazyfactory/tslint-rules/lib"
+                ],
+                "rules": {
+                    "align": [
+                        true,
+                        "parameters",
+                        "arguments",
+                        "statements"
+                    ],
+                    "chai-vague-errors": true,
```
Comment (cyberhck) [inline .eslintrc.js]: I think chai doesn't need to be there for us? I think it's when you pass the error message yourself, but jest doesn't allow that

**Comment by paibamboo** [comment]
By running
```
npx tslint-to-eslint-config
```
For any TSLint rules with corresponding ESLint equivalents, those equivalents will be used in the new configuration. TSLint rules without ESLint equivalents will be wrapped with eslint-plugin-tslint.

So old rules from `tslint-rules` and `tslint:react` are still applied.

File: .eslintrc.js (hunk 1 above)
Comment (paibamboo) [reply .eslintrc.js]: This is old rule from `tslint.json`

image

Ref: https://palantir.github.io/tslint/rules/one-line/

File: .eslintrc.js (hunk 2 above)
Comment (paibamboo) [reply .eslintrc.js]: Same, it's old rule we had in tslint.json. Yes, this can be removed.

**Comment by paibamboo** [comment]
> we can't extend with the things we were extending before? (tslint:latest and tslint-react) I think aren't applied

They are applied.

> were ALL rules available? If not we should make a list and watch them, once they're available we should probably add them as well

All old rules are applied.

> couldn't we have used both tslint and eslint and slowly move rules gradually? That'd mean all rules would still be in effect.

I think this is what we are doing. For any TSLint rules with corresponding ESLint equivalents, those equivalents will be used in the new configuration. TSLint rules without ESLint equivalents will be wrapped with eslint-plugin-tslint.

> what's gonna happen to the rules we wrote in tslint-rules?

They are still applied.

**Comment by cyberhck** [comment]
I meant this: https://github.com/crazyfactory/tslint-rules they're also applied? Wow, awesome, didn't know they just work, will approve.

**Pull request merged**